staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/tableconvertor/tableconvertor_test.go

```diff
@@ -201,6 +202,122 @@ func Test_convertor_ConvertToTable(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "Return table with additional column containing multiple string array values",
+			fields: fields{
+				headers: []metav1.TableColumnDefinition{
+					{Name: "name", Type: "string"},
+					{Name: "hosts", Type: "string"},
+				},
+				additionalColumns: []*jsonpath.JSONPath{
+					newJSONPath("hosts", "{.spec.servers[*].hosts}"),
+				},
+			},
+			args: args{
+				obj: &unstructured.Unstructured{
+					Object: map[string]interface{}{
+						"apiVersion": "example.istio.io/v1alpha1",
+						"kind":       "Blah",
+						"metadata": map[string]interface{}{
+							"name": "blah",
+						},
+						"spec": map[string]interface{}{
+							"servers": []map[string]interface{}{
+								{"hosts": []string{"foo", "bar"}},
+								{"hosts": []string{"baz"}},
+							},
+						},
+					},
+				},
+				tableOptions: nil,
+			},
+			want: &metav1.Table{
+				ColumnDefinitions: []metav1.TableColumnDefinition{
+					{Name: "name", Type: "string"},
+					{Name: "hosts", Type: "string"},
+				},
+				Rows: []metav1.TableRow{
+					{
+						Cells: []interface{}{"blah", `["foo","bar"] ["baz"]`},
+						Object: runtime.RawExtension{
+							Object: &unstructured.Unstructured{
+								Object: map[string]interface{}{
+									"apiVersion": "example.istio.io/v1alpha1",
+									"kind":       "Blah",
+									"metadata": map[string]interface{}{
+										"name": "blah",
+									},
+									"spec": map[string]interface{}{
+										"servers": []map[string]interface{}{
+											{"hosts": []string{"foo", "bar"}},
+											{"hosts": []string{"baz"}},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "Return table with additional column containing multiple integer array values",
+			fields: fields{
+				headers: []metav1.TableColumnDefinition{
+					{Name: "name", Type: "string"},
+					{Name: "bars", Type: "integer"},
+				},
+				additionalColumns: []*jsonpath.JSONPath{
+					newJSONPath("bars", "{.spec.foo[*].bar}"),
+				},
+			},
+			args: args{
+				obj: &unstructured.Unstructured{
+					Object: map[string]interface{}{
+						"apiVersion": "example.istio.io/v1alpha1",
+						"kind":       "Blah",
+						"metadata": map[string]interface{}{
+							"name": "blah",
+						},
+						"spec": map[string]interface{}{
+							"foo": []map[string]interface{}{
+								{"bar": []int64{1, 2}},
+								{"bar": []int64{3}},
+							},
+						},
+					},
+				},
+				tableOptions: nil,
+			},
+			want: &metav1.Table{
+				ColumnDefinitions: []metav1.TableColumnDefinition{
+					{Name: "name", Type: "string"},
+					{Name: "bars", Type: "integer"},
+				},
+				Rows: []metav1.TableRow{
+					{
+						Cells: []interface{}{"blah", `[1,2] [3]`},
```

Comment:
What I was saying in the other comment is that I didn't expect a string here for the second cell.  It should be `[[1,2],[3]]`, not a string.  But to do that properly, we have verify that clients properly handle it (in practice) even if in theory they are required to handle cells they don't understand.

Reply:
OK, I think I see what you're saying.  Let me take another look.

Reply:
Thank you for taking the time to review.  I made some changes that I think address your comments, separating the concerns between the TableConverter (server-side) and TablePrinter (client-side, responsible for formatting the output).  Please take another look when you get a chance.

Reply:
Seems I have a failing etcd unit test now... I'll need to see what is going on with that